tasks.py

```diff
@@ -1089,42 +1098,56 @@ def ci_build(context):
 
-                        # ci_build_arch(temp_dir, arch, build["commit_id"], include_private_scenes)
-                        p = multiprocessing.Process(target=ci_build_arch, args=(temp_dir, arch, build["commit_id"], include_private_scenes,))
-                        active_procs = lambda x: sum([p.is_alive() for p in x])
-                        started = False
-                        for _ in range(200):
-                            if active_procs(procs) > 0:
-                                logger.info("too many active procs - waiting before start %s " % arch)
-                                time.sleep(15)
-                                continue
-                            else:
-                                logger.info("starting build process for %s " % arch)
-                                started = True
-                                p.start()
-                                # wait for Unity to start so that it can pick up the GICache config
-                                # changes
-                                time.sleep(30)
-                                procs.append(p)
-                                break
-                        if not started:
-                            logger.error("could not start build for %s" % arch)
+                        build_success_queue = multiprocessing.Queue()
+                        p = multiprocessing.Process(
+                            target=ci_build_arch,
+                            kwargs=dict(
+                                root_dir=temp_dir,
+                                arch=arch,
+                                commit_id=build["commit_id"],
+                                build_success_queue=build_success_queue,
+                                include_private_scenes=include_private_scenes,
+                                immediately_fail_and_push_log=has_any_build_failed, # Don't bother trying another build if one has already failed
+                            )
+                        )
+                        p.start()
+
+                        build_success = False
+                        minutes_to_wait = 120
+                        for i in range(minutes_to_wait):
+                            try:
+                                build_success = build_success_queue.get(timeout=60)
+                            except (TimeoutError, Empty):
+                                logger.info(f"Waiting for build to complete, have waited {i+1} minutes ({minutes_to_wait} max).")
+
+                        has_any_build_failed = has_any_build_failed or not build_success
+                        if build_success:
+                            logger.info(
+                                f"Build success detected for {arch} {build['commit_id']}"
+                            )
+                        else:
+                            logger.error(f"Build failed for {arch}")
+                            p.kill()
+
+                        p.join(10.0)
```

Comment:
Is this a 10 second timeout, is this enough?

Reply:
I think it's enough but we can increase if it ends up being too short.